### ISSUE TITLE
Jet Force Gemini: Renders at low resolution when enabling two required frame buffer emulation settings

### ISSUE THREAD

**Issue by Mushman** [gonetz/GLideN64#378]
Jet Force Gemini needs two different frame buffer setting enabled to show particular effects. Examples of each are found in the character select screen.

Copy frame buffer to RDRAM / EnableCopyColorToRDRAM is needed for the character monitor to display.

Setting on
![The character monitor shows if the frame buffer is copied to the RDRAM](https://cloud.githubusercontent.com/assets/6550546/7085904/aa89063c-dfbe-11e4-8dbe-828c25ba1aa6.png)

Setting off
![Since the RDRAM is empty, the monitor will be blank with the setting off](https://cloud.githubusercontent.com/assets/6550546/7086000/90c3b19c-dfbf-11e4-9aff-72b0449829b2.png)

Render frame buffer as texture / EnableCopyColorFromRDRAM gets the number overlays for the characters' internal scan to appear.

Enabled
![When rendering the frame buffer to a texture, the overlay numbers for the internal scan can be seen](https://cloud.githubusercontent.com/assets/6550546/7086149/f8cd9996-dfc0-11e4-8977-5b90045b7fbe.png)

Disabled
![If the setting is off, they're hidden](https://cloud.githubusercontent.com/assets/6550546/7086195/5be2855a-dfc1-11e4-9012-806ab5da3702.png)

However, when both options are set at the same time, the game renders at a low resolution.

![Even though you can't see both effects simultaneously ingame, this is how it looks with both settings](https://cloud.githubusercontent.com/assets/6550546/7086240/c8ddadf6-dfc1-11e4-97a8-8c8950181850.png)

This doesn't always happen when these settings are used in other games.

![Putting on both settings acts find in some circumstances, like Goldeneye](https://cloud.githubusercontent.com/assets/6550546/7086737/b8f9353a-dfc7-11e4-94cc-cc0bd3874b2c.png)

Behaviour occurs in all versions of GLideN64 on both specifications regardless of other settings.

**Comment by gonetz** [comment]
Well, low-res image in JFG is actually correct. On first step plugin copies hi-res image into low-res color buffer in RDRAM. On second step plugin uses that low-res image as full screen texture. This image also contains the number, which CPU copied there. The result is expected. I need to think, is it possible to separate data written by CPU from the data copied by the plugin.

**Comment by Mushman** [comment]
I had actually noticed this behaviour some time ago, but didn't report until now since other games didn't need those settings and I wasn't sure whether it was expected. Thanks for clarifying.

**Comment by purplemarshmallow** [comment]
Conker's BFD also needs render frame buffer as texture option in uga buga level but it's in low-res with this option - same problem

**Comment by gonetz** [comment]
Yes. "It's war" cut-scene also looks correct, that is monochrome, only with both options on.

**Comment by purplemarshmallow** [comment]
Correct with Rice Video!
It needs the "with emulator" fb option and 1964.

**Comment by gonetz** [comment]
https://www.youtube.com/watch?v=Jp4q2BAMY5s

**Comment by duke912** [comment]
2022 and the low res still persists, Tawfret goes from 4k to 240p in raining areas, and also on Floyd mini-games. Will it be fixed? or at least a hack or something to disable these effects without getting a black screen?

**Comment by Jj0YzL5nvJ** [comment]
https://github.com/gonetz/GLideN64/issues/2653#issuecomment-1053396453

**Comment by duke912** [comment]
Is there a fix for PJ64?

**Comment by weinerschnitzel** [comment]
I doubt there will be soon. Last I recall, zilmar tried adding the fbinfo hooks to pj64 but there was a significant performance hit. 

I'm not sure if that meant pj64's dynarec became unplayable, but I would guess that you have to come up with a clever way to handle the reads/writes in the dynarec without having to interpret instructions that interact with the framebuffer.

**Comment by duke912** [comment]
I'm playing on mupen64plus-next retroarch, the same problem persists there, where can I find that 'fbinfo' whatever option? there's no option to enable it, and I couldn't find anything in the ini files...

**Comment by Jj0YzL5nvJ** [comment]
You just have to switch to `interpreter` (don't use `dynarec`)...

> I'm playing on mupen64plus-next retroarch, the same problem persists there, where can I find that 'fbinfo' whatever option? there's no option to enable it, and I couldn't find anything in the ini files...

### Mupen64Plus-Next:

> [Specifies INI Settings behaviour. This should really only contain essential options. Changing this can and will break ROM's, if the correct options aren't set manually. Some options may only be set via INI (fbInfoDisabled).](https://github.com/libretro/mupen64plus-libretro-nx/blob/6e9dcd2cd9d23d3e79eaf2349bf7e9f25ad45bf1/libretro/libretro_core_options.h#L999)

> ["[JET%20FORCE%20GEMINI]\n"
"Good_Name=Jet Force Gemini (E)(U)\n"
"frameBufferEmulation\\fbInfoDisabled=0\n"
"frameBufferEmulation\\copyAuxToRDRAM=1\n"](https://github.com/libretro/mupen64plus-libretro-nx/blob/fa04c66f8c6dec6af59cd72971237553d4b0b20b/custom/GLideN64/GLideN64.custom.ini.h#L124-L127)

### GLideN64:

> [[JET%20FORCE%20GEMINI]
Good_Name=Jet Force Gemini (E)(U)
frameBufferEmulation\fbInfoDisabled=0
frameBufferEmulation\copyAuxToRDRAM=1
frameBufferEmulation\copyToRDRAM=0](https://github.com/gonetz/GLideN64/blob/c8b3f6378b3d41b3f39d4d5ed0621dcb860c4b9e/ini/GLideN64.custom.ini#L128-L132)

**Comment by CtrlP1** [comment]
> You just have to switch to `interpreter` (don't use `dynarec`)...
> 
> > I'm playing on mupen64plus-next retroarch, the same problem persists there, where can I find that 'fbinfo' whatever option? there's no option to enable it, and I couldn't find anything in the ini files...
> 
> ### Mupen64Plus-Next:
> 
> > [Specifies INI Settings behaviour. This should really only contain essential options. Changing this can and will break ROM's, if the correct options aren't set manually. Some options may only be set via INI (fbInfoDisabled).](https://github.com/libretro/mupen64plus-libretro-nx/blob/6e9dcd2cd9d23d3e79eaf2349bf7e9f25ad45bf1/libretro/libretro_core_options.h#L999)
> 
> > ["[JET%20FORCE%20GEMINI]\n"
> > "Good_Name=Jet Force Gemini (E)(U)\n"
> > "frameBufferEmulation\fbInfoDisabled=0\n"
> > "frameBufferEmulation\copyAuxToRDRAM=1\n"](https://github.com/libretro/mupen64plus-libretro-nx/blob/fa04c66f8c6dec6af59cd72971237553d4b0b20b/custom/GLideN64/GLideN64.custom.ini.h#L124-L127)
> 
> ### GLideN64:
> 
> > [[JET%20FORCE%20GEMINI]
> > Good_Name=Jet Force Gemini (E)(U)
> > frameBufferEmulation\fbInfoDisabled=0
> > frameBufferEmulation\copyAuxToRDRAM=1
> > frameBufferEmulation\copyToRDRAM=0](https://github.com/gonetz/GLideN64/blob/c8b3f6378b3d41b3f39d4d5ed0621dcb860c4b9e/ini/GLideN64.custom.ini#L128-L132)

While this fixed most of the weapons reticle, it breaks the snipers. It also breaks any maps with rain and some cut scenes have contain visual glitches.
